### PR TITLE
Feature/profile page

### PR DESCRIPTION
**Pull request by dalbanhi**
## New Features Implemented
- [x] implemented displaying user profile page
- [x] implemented loading existing information onto profile/update form page
- [x] added location to user profile info

Here is a link to it working: [Loom video working](https://www.loom.com/share/b8266f809fa44004b566ddbb5489f0b5?sid=bd7508fd-e925-43b8-bd8e-a7489bc5edc3)

### Known bugs
- If Enter is pressed when selecting a location, an error is fired, even though later information is updated correctly. it might be a rendering issue somehow?
- If the Tech stack selector is not edited, there is an error with the form and it will not submit. The TechStackSelector's onChange function seems to set the selectedOptions to be the value for the techStack field correctly, but otherwise, the resolver will say that each element is the wrong type. After many attempt to fix it, no solution was yet found. 

Here is a video of the bugs: [Bugs Reproduction](https://www.loom.com/share/f28ee56493994f98a98437aa20295cb9?sid=8b132d89-5b1d-447d-8134-b96c7800db00)

### CODE REVIEW

File: src/components/shared/RightSidebar/RightSidebar.tsx
Comment (brandonetter) [inline src/components/shared/RightSidebar/RightSidebar.tsx]:
```suggestion

import React from "react";
import ProfileLink from "./ProfileLink";
import TagsList from "./TagsList";
import SocialLinks from "./SocialLinks";

const RightSidebar = () => {
  return (
    <aside className="flex min-h-screen w-3/12 flex-col gap-2 bg-myBlack-800 p-4">
      <ProfileLink />
      <TagsList />
      <SocialLinks />
    </aside>
  );
};

export default RightSidebar;
```
Since we're using the Social Links and the TagList to determine what's rendered, these commented out sections can be removed.

File: src/components/shared/Profile/KnowledgeLevelPreview.tsx
Comment (brandonetter) [inline src/components/shared/Profile/KnowledgeLevelPreview.tsx]:
```suggestion
```

File: src/components/shared/Form/Update/Profile/UpdateProfileForm.tsx
Comment (brandonetter) [inline src/components/shared/Form/Update/Profile/UpdateProfileForm.tsx]: ## This Addresses an Error Noted in Your Video:

Since the techStack from our database is just an array- we can map over it- which is a little simpler.

The next issue here will be that our form and form schema requires a label as well as a value:

```js
  const techStackToPass = (userFromDB?.techStack).map((item) => {
    return { label: item, value: item };
  });
```

The reason you were able to circumvent the error by adding a value is likely because the "setting" is done correctly and probably fixed our default values to some degree when the were added to the form.

:)

File: src/components/shared/Form/Update/Profile/UpdateProfileForm.tsx
Comment (brandonetter) [inline src/components/shared/Form/Update/Profile/UpdateProfileForm.tsx]:
```suggestion
  const knowledgeLevelsToPass = (userFromDB?.knowledgeLevels).map((item) => {
    return { value: item };
  });
```
Same thing here, we can simplify the way we're mapping the data if the data is already an array

File: src/components/shared/ContributionsTracker.tsx
Comment (brandonetter) [inline src/components/shared/ContributionsTracker.tsx]: This was mostly as a demo for the tooltip itself- so this is my bad. But at the moment it doesn't make much sense for this section to just say "hover" as a tooltip.

File: src/components/shared/Form/Onboarding/Steps/Steps.tsx
Comment (brandonetter) [inline src/components/shared/Form/Onboarding/Steps/Steps.tsx]: Regarding the error in your video:

## Address
Pressing "Enter" fires the 'onPlaceSelected' event with whatever the user typed instead of with a selection. If you look at the console.log you have for 'place':
```js
  onPlaceSelected={(place) => {
                console.log(place);
                onChange(place.formatted_address);
              }}
```
You can see this log appear:
image

We get an object without formatted_address, but instead just a `name` key.

The problem is that we're grabbing the "place.formatted_address", but not handling the case where the user typed a location manually.

We can address this by falling back to the "place.name" if "formatted_address" doesn't exist.

```js
  onChange(place?.formatted_address || place?.name);
 ```

## Enter Button
We also have an issue where pressing the `Enter` key inside this form element is submitting buttons. Because it's all in the same form- the Enter button is being used as a shortcut to "submit".

We can prevent the default behavior here by adding an "onKeyDown" event and, if the key is the 'Enter' key, stopping the default behavior:


```js
 <Autocomplete
         ...
              onKeyDown={(e) => {
                if (e.key === "Enter") {
                  e.preventDefault();
                  e.stopPropagation();
                }
              }}
 ```

File: src/components/shared/Form/Onboarding/Steps/Steps.tsx
Comment (brandonetter) [inline src/components/shared/Form/Onboarding/Steps/Steps.tsx]: Once we've solved this issue, let's remove the console log from testing

File: src/components/shared/Form/Update/Profile/UpdateProfileForm.tsx
Comment (brandonetter) [inline src/components/shared/Form/Update/Profile/UpdateProfileForm.tsx]:
```suggestion
```
Let's remove this testing log once we're done here.

If we keep all of these logs in, by the time we're done with the project we'll have a lot of trouble debugging any issue!

File: src/components/shared/Form/Onboarding/Steps/Steps.tsx
Comment (brandonetter) [inline src/components/shared/Form/Onboarding/Steps/Steps.tsx]:
```suggestion
```
Let's remove this log from our controller render

**Review by brandonetter — CHANGES_REQUESTED** [review]
This is all coming together great!

And the size of this PR is a bit misleading- over half of this is shadcn, icons, and file movement. No worries :)

Let me know if there's any help you need implementing these changes or any questions about why I've made any suggestions.
🚀

File: src/components/shared/Form/Update/Profile/UpdateProfileForm.tsx
Comment (dalbanhi) [reply src/components/shared/Form/Update/Profile/UpdateProfileForm.tsx]: Ah, I see. That makes sense. Thanks!

File: src/components/shared/Form/Onboarding/Steps/Steps.tsx
Comment (dalbanhi) [reply src/components/shared/Form/Onboarding/Steps/Steps.tsx]: Very helpful, and that makes sense. I had only encountered stopPropagation a few times before, but this makes a lot of sense!

**Comment by dalbanhi** [comment]
Hey Brandon, thanks for your very helpful feedback; I just implemented the changes you requested!

**Review by brandonetter — APPROVED** [review]
Great! Assuming everything is working- I do have a final question before we merge this.

Are you getting fresh data on the "profile" page after you edit in the "update" page?

For example, if you add information or change the name etc.. in the edit profile page- do you always see updated versions of that data when sent back to the view profile page?

If not- we can talk a bit about revalidating data and caching! But if all the bugs are worked out and you're not getting that behavior- I think we're good to merge unless you have any final questions about this section

**Comment by dalbanhi** [comment]
Hey Brandon, yes, I have seen it in almost all cases. There was one case in which I did have to refresh the page-- so I think talkinga bout caching would be useful!

**Pull request merged**
